Sources/Swiftly/Update.swift

```diff
@@ -31,61 +38,154 @@ struct Update: SwiftlyCommand {
+                return
+            }
+        }
+
+        try await Install.execute(version: newToolchain, &config, self.httpClient)
+
+        if config.inUse == parameters.oldToolchain {
+            try await Use.execute(newToolchain, &config)
+        }
+
+        try await Uninstall.execute(parameters.oldToolchain, &config)
+        SwiftlyCore.print("Successfully updated \(parameters.oldToolchain) ⟶ \(newToolchain)")
     }
 
-    private func oldToolchain() throws -> ToolchainVersion? {
-        guard let input = self.toolchain else {
-            return try Config.load().inUse
+    private func resolveUpdateParameters(_ config: Config) throws -> UpdateParameters? {
+        let selector = try self.toolchain.map { try ToolchainSelector(parsing: $0) }
+
+        let oldToolchain: ToolchainVersion?
+        if let selector {
+            let toolchains = config.listInstalledToolchains(selector: selector)
+            // When multiple toolchains are matched, update the latest one.
+            // This is for situations such as `swiftly update 5.5` when both
+            // 5.5.1 and 5.5.2 are installed (5.5.2 will be updated).
+            oldToolchain = toolchains.max()
+        } else {
+            oldToolchain = config.inUse
         }
 
-        let selector = try ToolchainSelector(parsing: input)
-        let toolchains = try Config.load().listInstalledToolchains(selector: selector)
+        guard let oldToolchain else {
+            return nil
+        }
 
-        // When multiple toolchains are matched, update the latest one.
-        // This is for situations such as `swiftly update 5.5` when both
-        // 5.5.1 and 5.5.2 are installed (5.5.2 will be updated).
-        return toolchains.max()
+        switch oldToolchain {
+        case let .snapshot(snapshot):
+            return .snapshot(old: snapshot)
+        case let .stable(stable):
+            switch selector {
+            case .none:
+                return .stable(old: stable, target: .latestPatch)
+            case let .stable(_, minor, _):
+                if minor == nil {
+                    return .stable(old: stable, target: .latestMinor)
+                } else {
+                    return .stable(old: stable, target: .latestPatch)
+                }
+            case .latest:
+                return .stable(old: stable, target: .latest)
+            default:
+                fatalError("unreachable")
+            }
+        }
     }
 
-    private func newToolchain(old: ToolchainVersion) async throws -> ToolchainVersion? {
-        switch old {
-        case let .stable(oldRelease):
-            return try await HTTP.getReleaseToolchains(limit: 1) { release in
-                release.major == oldRelease.major
-                    && release.minor == oldRelease.minor
-                    && release.patch > oldRelease.patch
+    private func fetchNewToolchain(_ bounds: UpdateParameters) async throws -> ToolchainVersion? {
```

Comment:
Can we function header comments here and on other functions. Is this downloading the toolchain?